### PR TITLE
Ported minibuffer-selected-window

### LINKED ISSUE #326: Port minibuffer-selected-window to Rust
```c
DEFUN ("minibuffer-selected-window", Fminibuffer_selected_window, Sminibuffer_selected_window, 0, 0, 0,
       doc: /* Return the window which was selected when entering the minibuffer.
Returns nil, if selected window is not a minibuffer window.  */)
  (void)
{
  if (minibuf_level > 0
      && MINI_WINDOW_P (XWINDOW (selected_window))
      && WINDOW_LIVE_P (minibuf_selected_window))
    return minibuf_selected_window;

  return Qnil;
}
```

### PR DESCRIPTION
**Pull request by Yarikx**
Closes #326

### CODE REVIEW

File: rust_src/src/windows.rs [hunk 1]
```diff
@@ -135,3 +136,18 @@ pub fn window_minibuffer_p(window: LispObject) -> LispObject {
...
+
+/// Return the window which was selected when entering the minibuffer.
+/// Returns nil, if selected window is not a minibuffer window.
+#[lisp_fn]
+pub fn minibuffer_selected_window() -> LispObject {
+    let level = unsafe { minibuf_level };
+    let current_minibuf = unsafe { LispObject::from_raw(current_minibuf_window) };
+    if level > 0 && selected_window().as_window_or_error().is_minibuffer() &&
+        current_minibuf.as_window_or_error().is_live()
+    {
```
Comment (brotzeit) [inline rust_src/src/windows.rs]: I think the c implementation isn't throwing errors here.

Comment (brotzeit) [inline rust_src/src/windows.rs]: I think the c implementation isn't throwing errors here.

Comment (brotzeit) [inline rust_src/src/windows.rs]: I think the c implementation does not throw errors here.

Comment (brotzeit) [inline rust_src/src/windows.rs]: I think you shouldn't raise errors here.

Comment (birkenfeld) [reply rust_src/src/windows.rs]: The alternative would be `unwrap()`... not sure that is better.

Comment (brotzeit) [reply rust_src/src/windows.rs]: I don't think it's better, but it looks like the c code does not throw errors here.

Comment (Yarikx) [reply rust_src/src/windows.rs]: Makes sense. As only windows object can be in `current_minibuf` static var, an error should not occur there. I think we can safely unwrap().

**Pull request merged**